src/archive/src/main.rs

```diff
@@ -239,6 +259,84 @@ fn write_entry(anchor: AnchorNumber, timestamp: Timestamp, entry: ByteBuf) {
             ))
         }
     });
+    write_entry_internal(anchor_number, timestamp, entry)
+}
+
+/// Fetches, archives and acknowledges a batch of entries.
+/// *Note:* Must be written in a way that nothing breaks on overlapping executions of [fetch_entries].
+async fn fetch_entries() {
+    let ii_canister = with_config(|config| config.ii_canister);
+    let fetch_result: CallResult<(Vec<BufferedEntry>,)> =
+        call(ii_canister, "fetch_entries", ()).await;
+
+    let entries = match fetch_result {
+        Ok((entries,)) => entries,
+        Err((code, message)) => {
+            // failed to fetch entries --> store failure information and exit early
+            store_call_error(CallErrorInfo {
+                time: time(),
```

Comment:
nit: how about storing the time when the call was made? if the call e.g. times out this value may be misleading